pkg/motor/create_account.go

```diff
@@ -179,9 +166,29 @@ func pskEncrypt(shard []byte) ([]byte, []byte, error) {
 	return cipherShard, key, nil
 }
 
-func dscEncrypt(shard, dsc []byte) ([]byte, error) {
-	if len(dsc) != 32 {
-		return nil, errors.New("dsc must be 32 bytes")
+// dscEncrypt encrypts the shard with the DSC key
+// Returns: encrypted shard, given key, error
+func dscEncrypt(shard, d []byte) ([]byte, []byte, error) {
+	// Check if the DSC is valid
+	if len(d) != 32 {
```

Comment:
This creates unexpected behavior. If the caller provides a key, they'll expect that to be used.